flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/format/cdc/CDCSchemaAdapter.java

```diff
@@ -111,28 +110,13 @@ public boolean equals(Object rhs) {
    * Number of physical fields.
    */
   private final int physicalFieldCount;
+  private final DataType physicalDataType;
 
-  /**
-   * Number of final produced fields.
-   */
-  private final int producedFieldCount;
-
-  public CDCSchemaAdapter(final DataType physicalDataType,
-      final Function<DataType, TypeInformation<RowData>> typeInfoFactory,
-      @Nullable final CDCMetadata[] metadata) {
+  public CDCSchemaAdapter(final DataType physicalDataType, CDCMetadata[] metadata) {
     this.metadata = CDCMetadata.notNull(metadata);
+    this.physicalDataType = physicalDataType;
```

Comment:
What is a producedField?

Reply:
There are some different datatypes in flink api:
1. produced type, it tells downstream types for RowData;
2. physicalDataType, real types, exclude meta columns and computed columns;
3. all types, the types contains all of meta/physical/computed columns.